src/omnisharp/OmnisharpDownloader.ts

```diff
@@ -0,0 +1,25 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import * as utils from '../common';
+import * as fs from 'fs';
+
+export class OmnisharpDownloader {
+
+    public GetLatestExperimentVersion(): string {
+        let basePath = path.resolve(utils.getExtensionPath(), ".omnisharp/experiment");
+        let compareVersions = require('compare-versions');
+        let latestVersion: string;
+        let items = fs.readdirSync(basePath);
+        if (items) {
+            items.sort(compareVersions);
```

Comment:
This could just be a for loop (or whatever JS/TS equivalent there is to Enumerable.Min).

Reply:
probably better to do a loop and use package like semver to address also the non-semantic version string (this will return a null object) and also handling pre-release versions